osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -39,26 +43,47 @@ public class SkinProvidingContainer : Container, ISkinSource
 
         protected virtual bool AllowColourLookup => true;
 
+        /// <summary>
+        /// Constructs a new <see cref="SkinProvidingContainer"/> with a single skin added to the protected <see cref="SkinSources"/> list.
+        /// </summary>
         public SkinProvidingContainer(ISkin skin)
+            : this()
         {
-            this.skin = skin;
+            SkinSources.Add(skin);
+        }
 
+        /// <summary>
+        /// Constructs a new <see cref="SkinProvidingContainer"/> with no sources.
+        /// Up to the implementation for adding to the <see cref="SkinSources"/> list.
+        /// </summary>
+        protected SkinProvidingContainer()
+        {
             RelativeSizeAxes = Axes.Both;
 
             noFallbackLookupProxy = new NoFallbackProxy(this);
 
-            if (skin is ISkinSource source)
-                source.SourceChanged += TriggerSourceChanged;
+            SkinSources.BindCollectionChanged(((_, args) =>
+            {
+                switch (args.Action)
+                {
+                    case NotifyCollectionChangedAction.Remove:
+                        foreach (var source in args.OldItems.Cast<ISkin>().OfType<ISkinSource>())
+                            source.SourceChanged -= OnSourceChanged;
+
+                        break;
+
+                    case NotifyCollectionChangedAction.Add:
+                        foreach (var source in args.NewItems.Cast<ISkin>().OfType<ISkinSource>())
+                            source.SourceChanged += OnSourceChanged;
+
+                        break;
+                }
+            }), true);
```

Comment:
I'm not sure if there will still be `SkinProvidingContainer`s with `ISkinSource`s going forward (transformers now are `ISkin`s), but I wasn't sure about just throwing this logic away.